src/wp-includes/block-supports/layout.php

```diff
@@ -33,24 +33,51 @@ function wp_register_layout_support( $block_type ) {
  * Generates the CSS corresponding to the provided layout.
  *
  * @since 5.9.0
+ * @since 6.1.0 Added `$block_spacing` param, use style engine to enqueue styles.
  * @access private
  *
  * @param string  $selector                      CSS selector.
  * @param array   $layout                        Layout object. The one that is passed has already checked
  *                                               the existence of default block layout.
- * @param boolean $has_block_gap_support         Whether the theme has support for the block gap.
+ * @param bool    $has_block_gap_support         Whether the theme has support for the block gap.
  * @param string  $gap_value                     The block gap value to apply.
- * @param boolean $should_skip_gap_serialization Whether to skip applying the user-defined value set in the editor.
- * @param string  $fallback_gap_value            The custom fallback value for block gap.
+ * @param bool    $should_skip_gap_serialization Whether to skip applying the user-defined value set in the editor.
+ * @param string  $fallback_gap_value            The block gap value to apply.
+ * @param array   $block_spacing                 Custom spacing set on the block.
  * @return string CSS style.
```

Comment:
Add empty string in return document.
```suggestion
 * @return string CSS style or an empty string.
```